R/tufte_handout.R

```diff
@@ -56,26 +46,50 @@ tufte_handout <- function(fig_width = 4,
   
   # set hooks for special plot output
   knitr_options$knit_hooks$plot <- function(x, options) {
-    
-    # determine caption (if any)
-    caption <- ifelse(is.null(options$fig.cap), 
-                      "",
-                      paste("\\caption{", options$fig.cap, "}\n", sep = ""))
-    
+
     # determine figure type
     if (isTRUE(options$fig.margin)) 
-      figtype <- "marginfigure"
+      options$fig.env <- "marginfigure"
     else if (isTRUE(options$fig.fullwidth))
-      figtype <- "figure*"
-    else
-      figtype <- "figure"
-    
-    # return the latex
-    paste(sprintf('\\begin{%s}\n \\includegraphics{%s}\n%s\\end{%s}\n',
-                  figtype, x, caption, figtype))
+      options$fig.env <- "figure*"
+
+    knitr::hook_plot_tex(x, options)
   }
   
   # override the knitr settings of the base format and return the format
   format$knitr <- knitr_options
   format
 }
+
+#' @details \code{tufte_html()} provides the HTML format based on the Tufte CSS:
+#'   \url{https://edwardtufte.github.io/tufte-css/}.
+#' @param css One or more CSS files to include, and the Tufte CSS will always be
+#'   included in this list
+#' @rdname tufte_handout
+#' @export
+tufte_html <- function(css = NULL, ...) {
+  tufte_css <- rmarkdown_system_file(
+    'rmarkdown', 'templates', 'tufte_html', 'resources', 'tufte.css'
+  )
+  css <- c(tufte_css, css)
+  html_document(css = css, theme = NULL, ...)
+}
+
+#' @details \code{newthought()} can be used in inline R expressions in R
+#'   Markdown (e.g. \samp{`r newthought(Some text)`}), and it works for both
+#'   HTML (\samp{<span class="newthought">text</span>}) and PDF
+#'   (\samp{\\newthought{text}}) output.
+#' @param text A character string to be presented as a \dQuote{new thought}
+#'   (using small caps)
+#' @rdname tufte_handout
+#' @export
+newthought <- function(text) {
+  if (is_html_output()) {
+    sprintf('<span class="newthought">%s</span>', text)
+  } else if (is_latex_output()) {
+    sprintf('\\newthought{%s}', text)
+  } else {
+    warning('newthought() only works for HTML and LaTeX output')
```

Comment:
I think it's good practice to do `call. = FALSE` in `warning()`, but I'm not sure what the RStudio style is.